mathesar/api/exceptions/error_codes.py

```diff
@@ -16,18 +16,23 @@ class ErrorCodes(Enum):
     # DB Error Codes
     DuplicateTableError = 4205
     DuplicateColumnError = 4206
+    ExclusionViolation = 4213
+    ForeignKeyViolation = 4212
     InvalidTypeCast = 4203
     InvalidTypeOption = 4210
     InvalidDefault = 4211
     NonClassifiedIntegrityError = 4201
     NotNullViolation = 4204
     RaiseException = 4202
+    TypeMismatchViolation = 4214
     UndefinedFunction = 4207
     UniqueViolation = 4208
     UnsupportedType = 4209
 
     # Data Imports error code
     InvalidTableError = 4301
+    NotNullImportViolation = 4302
+    UniqueImportViolation = 4303
```

Comment:
@mathemancer Can you please verify that these error codes are valid and can be added?

Reply:
As far as I can tell, these aren't conflicting with anything.